src/lex/cpp.rs

```diff
@@ -354,14 +356,16 @@ impl<'a> PreProcessor<'a> {
 
         let file_processor = FileProcessor::new(chars, filename, debug);
 
-        Self {
+        let mut new_cpp = Self {
             error_handler: Default::default(),
             nested_ifs: Default::default(),
             pending: Default::default(),
             search_path,
             definitions,
             file_processor,
-        }
+        };
+        new_cpp.update_builtin_definitions(); // So they are defined from the start
```

Comment:
Is there a reason this needs to be done eagerly instead of lazily? I think if you only call it just before any call to replace you should be fine.

Reply:
We do it once to ensure the builtins are considered defined for `#ifdef` which doesn't do any replacement.